### DISCUSSION TITLE
React-JSX, don't allow arrow functions as props

### DISCUSSION THREAD

**Discussion by VPKSoft** [biomejs/biome#4636] [Ideas]
Hi, 
I was hoping for a rule to mark arrow functions passed as JSX props as errors or warnings. This idea is based on the Q&A discussion [#4625](https://github.com/biomejs/biome/discussions/4625).
 In ESLint there this rule is called [jsx-no-bind](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md) but the implementation I'm suggesting is just to complain about arrow functions used as props:
**Wrong:**
```jsx
<button onClick={() => onOkClick()}>OK</button>
```
**Valid:**
```jsx
<button onClick={onOkClick}>OK</button>
```

So there would be no tracking of the prop origin, just the syntax.
Reason for this is that using arrow functions in JSX makes the code a bit messy and probably less reusable than just passing valid callbacks, functions, methods, etc.
*Sample image of the VS Code:*
![image](https://github.com/user-attachments/assets/2e9b4bdd-424f-4c35-8cee-779a144ab855)

**Comment by dyc3** [comment]
I've opened #4637 for this.

**Reply by VPKSoft** [reply]
Thank you.
